fix: TransferAsset vulnerability (#1134)
planetarium/libplanet: CHANGES.md

To be released.

 -  Fixed a vulnerability that `AccountStateDeltaImpl.TransferAsset()` if

>     sender and recipient are same. [[#1134]]

Author: https://github.com/planetarium/libplanet/pull/1134/commits/37b996fc3a7717d0853dbe077569279412e8c67d amended.

To be released.

 -  Fixed a vulnerability of the `IAccountStateDelta.TransferAsset()`'s internal
    implementation that it had doubled the balance for transferring to themselves.
    *Since this behavioral change breaks the protocol-level compatibility if any actions
    in your app had used `IAccountStateDelta.TransferAsset()`, you must make the action
    to conditionally work as it had done for the hard-coded block index before

>     this patch is applied, or you may be possible to hard-fork the network.*  [[#1134]]

Reviewer: It seems longer than 80 and it makes check `check-build` failed. 😢

Author: 74a5111866fcef81b4c94048c34ea67e69ff984b amended.

To be released.

 -  Fixed a vulnerability of the `IAccountStateDelta.TransferAsset()`'s
    internal implementation that it had doubled the balance for transferring
    to themselves. *Since this behavioral change breaks the protocol-level
    compatibility if any actions in your app had used
    `IAccountStateDelta.TransferAsset()`, you must make the action to
    conditionally work as it had done for the hard-coded block index before

>     this patch is applied, or you may be possible to
>     hard-fork the network.*  [[#1134]]

Reviewer 2: ```suggestion
    this patch is applied, or you should hard-fork the network.*
    [[#1134]]
```
